WooCommerce/Classes/ViewRelated/JetpackSetup/WPComLogin/WPComEmailLoginViewModel.swift

```diff
@@ -0,0 +1,97 @@
+import Combine
+import UIKit
+import WordPressShared
+
+/// View model for `WPComEmailLoginView`
+final class WPComEmailLoginViewModel: ObservableObject {
+    let titleString: String
+    let subtitleString: String
+
+    @Published var emailAddress: String = ""
+    /// Local validation on the email field.
+    @Published private(set) var isEmailValid: Bool = false
+
+    let termsAttributedString: NSAttributedString
+
+    private var emailFieldSubscription: AnyCancellable?
+
+    init(siteURL: String,
+         requiresConnectionOnly: Bool,
+         debounceDuration: Double = Constants.fieldDebounceDuration) {
+        self.titleString = requiresConnectionOnly ? Localization.connectJetpack : Localization.installJetpack
+        self.subtitleString = requiresConnectionOnly ? Localization.loginToConnect : Localization.loginToInstall
+        self.termsAttributedString = {
+            let content = String.localizedStringWithFormat(Localization.termsContent, Localization.termsOfService, Localization.shareDetails)
+            let paragraph = NSMutableParagraphStyle()
+            paragraph.alignment = .center
+
+            let mutableAttributedText = NSMutableAttributedString(
+                string: content,
+                attributes: [.font: UIFont.footnote,
+                             .foregroundColor: UIColor.secondaryLabel,
+                             .paragraphStyle: paragraph]
+            )
+
+            mutableAttributedText.setAsLink(textToFind: Localization.termsOfService,
+                                            linkURL: Constants.jetpackTermsURL + siteURL)
+            mutableAttributedText.setAsLink(textToFind: Localization.shareDetails,
+                                            linkURL: Constants.jetpackShareDetailsURL + siteURL)
+            return mutableAttributedText
+        }()
+        observeEmailField(debounceDuration: debounceDuration)
+    }
+}
+
+private extension WPComEmailLoginViewModel {
+    func observeEmailField(debounceDuration: Double) {
+        emailFieldSubscription = $emailAddress
+            .removeDuplicates()
+            .debounce(for: .seconds(debounceDuration), scheduler: DispatchQueue.main)
+            .sink { [weak self] email in
+                self?.validateEmail(email)
+            }
+    }
+
+    func validateEmail(_ email: String) {
+        isEmailValid = EmailFormatValidator.validate(string: email)
+    }
+}
+
+extension WPComEmailLoginViewModel {
+    private enum Constants {
+        static let fieldDebounceDuration = 0.3
+        static let jetpackTermsURL = "https://jetpack.com/redirect/?source=wpcom-tos&site="
+        static let jetpackShareDetailsURL = "https://jetpack.com/redirect/?source=jetpack-support-what-data-does-jetpack-sync&site="
+    }
+
+    enum Localization {
```

Comment:
nit: `private`

Reply:
I'm keeping this `internal` to reuse the strings in the view and tests.